Add git hooks and setup target in Makefile

syedaffanhamdani opened the pull request.

syedaffanhamdani committed:
Add git hooks and setup target in Makefile

cristim committed:
Merge branch 'master' into feat/add-pre-commit-hook

cristim commented on git-hooks/pre-commit:
```diff
@@ -0,0 +1,16 @@
+
+# Run Prettier on staged files
+echo "Running Prettier on staged files..."
+echo "$files" | xargs npx prettier --write
```
Can we also add the black linter here?

syedaffanhamdani replied:
added

cristim reviewed (changes requested):
Great contribution, this should help us catch the formatting issues much earlier.

Let's also document this in the README in the section for local development, so other people do this locally as well.

And while at it please also mention the tools they need to install.

syedaffanhamdani committed:
Merge remote-tracking branch 'leanercloud/master' into feat/add-pre-commit-hook

syedaffanhamdani committed:
precommit hook also formats changed python file and updated readme

syedaffanhamdani commented:
> Great contribution, this should help us catch the formatting issues much earlier.
> 
> Let's also document this in the README in the section for local development, so other people do this locally as well.
> 
> And while at it please also mention the tools they need to install.

updated readme and added black to requirements.txt file
